cpp-package/example/cpp-predictor/cpp_pred.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <string>
+#include <vector>
+#include <cassert>
+
+#include "mxnet-cpp/MxNetCpp.h"
+#include <opencv/cv.hpp>
+
+#include "predictor.h"
+
+int main() {
+    cv::Mat image = cv::imread("test.png");
+    image.convertTo(image, CV_32FC3, 1 / 255.);
+
+    mx_uint ch = image.channels();
+    mx_uint width = image.cols;
+    mx_uint height = image.rows;
+
+    size_t image_size = width * height;
+
+    std::vector<cv::Mat> rgb(3), dst(3);
+    cv::split(image, rgb);
+
+    mx_float *data = new mx_float[image_size * ch];
+    memcpy(data + image_size * 0, rgb[2].data, image_size * sizeof(mx_float));
+    memcpy(data + image_size * 1, rgb[1].data, image_size * sizeof(mx_float));
+    memcpy(data + image_size * 2, rgb[0].data, image_size * sizeof(mx_float));
+
+    try {
+        std::string symbol_json = "somesr2x_int8_quantized-symbol.json";
+        std::string param_data = "somesr2x_int8_quantized-0000.params";
+
+        auto ctx = Context::gpu(0);
```

Comment:
Generally we do it with
`ctx = mx.context.current_context()`

Curious to know why its set to GPU alone

Reply:
Since my workflow is very compute intensive, I usually leave CPU alone. LOL